### PR TITLE
Bump RCS version constants to 8.9

### PR DESCRIPTION
**Pull request by ywangd**
Since the feature is still behind feature flag. This PR bumps its version contants to the current version.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java [hunk 1]
```diff
@@ -38,8 +38,8 @@
  */
 public class RemoteClusterPortSettings {
 
-    public static final TransportVersion TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = TransportVersion.V_8_8_0;
-    public static final Version VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = Version.V_8_8_0;
+    public static final TransportVersion TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = TransportVersion.V_8_9_0;
+    public static final Version VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY = Version.V_8_9_0;
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java:42]: Please let me know if you want me to remove this constant in this PR. 
Relates: https://github.com/elastic/elasticsearch/pull/95442#discussion_r1173658588

**Review by n1v0lg — APPROVED** [review]

**Comment by ywangd** [comment]
@n1v0lg The BWC failure is a genuine serialization issue. We have following conditional code in the `NodeInfo` class:
https://github.com/elastic/elasticsearch/blob/b88f6684bd375a9c95dca42080edc48434db323b/server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java#L90-L92

Since the code is already there in 8.8, a 8.8 node already knows how to read/write this field. However, when the main bumps the version constant to be 8.9, it thinks 8.8 does not understand this field. Hence the mismatched expectation between 8.8 and 8.9 resulted into test failure. 

We have similar conditional serialization code in quite a few places, e.g. [PutRoleRequest](https://github.com/elastic/elasticsearch/blob/b88f6684bd375a9c95dca42080edc48434db323b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/PutRoleRequest.java#L62-L64), [RoleDescriptor](https://github.com/elastic/elasticsearch/blob/b88f6684bd375a9c95dca42080edc48434db323b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java#L182-L186), [SecurityFeatureSetUsage](https://github.com/elastic/elasticsearch/blob/b88f6684bd375a9c95dca42080edc48434db323b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/SecurityFeatureSetUsage.java#L78-L80), etc.

Most of them do not travel across nodes so we haven't seen this issue in past version constant bumps. But NodeInfo and `SecurityFeatureSetUsage` do cross nodes. 

Though it is possible to hand pick the places for the version bump, at this point, I think it is easier and safer to just introduce a new version constant for things developed in the 8.9 branch. This is also more accurate to the actual development progress. Concretely, I am suggesting:
* Rename `TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY` to `TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY_CCS` and keep it to be 8.8
* Add a new version constant `TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY_CCR` to be 8.9

What do you think? Sorry for the back and forth.

**Comment by n1v0lg** [comment]
@ywangd good catch! I'm good with your proposed approach. As you mention, we _technically_ have the same issue in other places, and with older versions (e.g., `PutRoleRequest`). However, since it doesn't manifest in practice and all relevant functionality sits behind a feature flag for those versions, I think there's value in using `TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY_CCS` everywhere we can and `TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY_CCR` for things we're introducing in 8.9. 

tl;dr I'm +1 on your proposed approach

File: server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java (hunk 1 above)
Comment (ywangd) [reply server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java]: I removed this constant to in favor of the transport one in this PR because I don't want to introduce yet another 8.9 Version constant to gate specialized API keys in ApiKeyService.

**Review by n1v0lg — APPROVED** [review]

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java [hunk 2]
```diff
@@ -312,7 +312,7 @@ private <T extends TransportResponse> void sendWithCrossClusterAccessHeaders(
                 }
                 final String remoteClusterAlias = remoteClusterCredentials.clusterAlias();
 
-                if (connection.getTransportVersion().before(TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY)) {
+                if (connection.getTransportVersion().before(TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY_CCR)) {
```
Comment (ywangd) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java:315]: Added this small but important adjustment to bump the version barrier to 8.9 for sending/receving cross-cluster requests because the new CCR actions are not understandable by 8.8 nodes.
